22891-Callbacks-with-more-than-4-parameters-in-Win64-is-not-working

tesonep opened the pull request:
Issue: https://pharo.manuscript.com/f/cases/22891/Callbacks-with-more-than-4-parameters-in-Win64-is-not-working

tesonep committed:
Fixing Alien callbacks to work in Win64 with more than 4 parameters

pavel-krivanek commented:
related test failures:
```
unix-32 / Tests-unix-32 / testFloatParameters – Unix32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
<1s
unix-32 / Tests-unix-32 / testMixingParameters – Unix32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
<1s
windows-32 / Tests-windows-32 / testFloatParameters – Windows32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
<1s
windows-32 / Tests-windows-32 / testMixingParameters – Windows32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
<1s
osx-32 / Tests-osx-32 / testFloatParameters – MacOSX32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
<1s
osx-32 / Tests-osx-32 / testMixingParameters – MacOSX32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
```

guillep commented:
yeah, we know. Pablo spent the afternoon on that yesterday, so we are
working on it :)

Le sam. 12 janv. 2019 à 10:48, pavel-krivanek <notifications@github.com> a
écrit :

> related test failures:
>
> unix-32 / Tests-unix-32 / testFloatParameters – Unix32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
> <1s
>
> unix-32 / Tests-unix-32 / testMixingParameters – Unix32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
> <1s
>
> windows-32 / Tests-windows-32 / testFloatParameters – Windows32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
> <1s
>
> windows-32 / Tests-windows-32 / testMixingParameters – Windows32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
> <1s
>
> osx-32 / Tests-osx-32 / testFloatParameters – MacOSX32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
> <1s
>
> osx-32 / Tests-osx-32 / testMixingParameters – MacOSX32.UnifiedFFI.Tests.Tests.FFICallbackParametersTests
>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/pharo-project/pharo/pull/2220#issuecomment-453734410>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AArO4q0kxKmu0HFRu_-mBQv47K6luBffks5vCa-IgaJpZM4Z7bc4>
> .
>
-- 



Guille Polito

Research Engineer

Centre de Recherche en Informatique, Signal et Automatique de Lille

CRIStAL - UMR 9189

French National Center for Scientific Research - *http://www.cnrs.fr
<http://www.cnrs.fr>*


*Web:* *http://guillep.github.io* <http://guillep.github.io>

*Phone: *+33 06 52 70 66 13

tesonep committed:
Refactoring the implementation of 32 bits callbacks.

tesonep committed:
- Adding more tests.
- Implementing for Unix 64 bits
- Improving the reuse of parts.

tesonep committed:
Fixing the size when it is a pointer.

tesonep committed:
Adding Win64 support

estebanlm commented:
what's the status?

tesonep committed:
Adding comments for all the classes and methods.
Documenting the methods and calling convention.

Also I am disabling the FFICallbackTests for windows, as there is still a bug in the VM.

tesonep commented:
> what's the status?

@estebanlm I will wait for the CI to test it in 32 bits and then I will try it in the 64 bits architecture, but with the generated images.
